### PR TITLE
Doxygen warnings

### PR DESCRIPTION
**Pull request by bdgackle**
When building doxygen docs in current master, there are 1,362 lines of warnings and errors generated.  This PR fixes all but one of them.

It also adds Doxygen documentation to the calendar class, and makes some trivial style and typo fixes in nearby code.

This should enable us to use the error output from Doxygen in the future to meaninfully track new errors, if desired, perhaps in an automated fashion.

### CODE REVIEW

**Comment by bdgackle** [comment]
I'm rolling back the changes to json.h because they are causing automated build failure, but I believe these changes to be correct.  I'll figure out how to go about making them later.

I don't want to leave the rest of this sitting longer than required, though, as I'm touching enough files here merge conflicts will become increasingly likely to emerge as time goes on.

**Comment by bdgackle** [comment]
So... same drill with json.cpp.

These two roll-back commits serve pretty well to show what the changes were.  Note that they only affect comments (added namespaces to some @ref tags, and added a couple of @relates tags).

**Comment by bdgackle** [comment]
On rereading, I think my last two comments may have suggested I was going to do more with those commits.  Just in case, and to clarify:

Both json.h/json.c are protected.  Builds auto-fail if they don't match master exactly, so I was removing any changes from them.  I don't plan on doing anything additional as part of this PR.  I think the changes themselves are good, but I'll submit them in a separate PR once I figure out the proper procedure for that.

Barring any comments (or outright rejection...), I'm not planning on further updates.  This is ready for review.

I'll keep an eye on it and periodically rebase to master if anything causes merge conflicts between now and whenever it gets reviewed.

File: src/vehicle.h [hunk 1]
```diff
@@ -138,7 +138,7 @@ struct vehicle_part : public JsonSerializer, public JsonDeserializer
      */
     float consume_energy( const itype_id &ftype, float energy );
 
-    /* Can part in current state be reloaded optionally with specific @ref obj */
+    /* @retun true if part in current state be reloaded optionally with specific itype_id */
```
Comment (Coolthulhu) [inline src/vehicle.h]: `@retun` - typo

**Comment by Coolthulhu** [comment]
> Both json.h/json.c are protected. Builds auto-fail if they don't match master exactly, so I was removing any changes from them.

The build failures are acceptable if the PR changes only those files.
As far as I know, the protection is just for safety, not something tied to licensing or anything like that.
You can PR the changes to those files separately.

**Pull request merged**